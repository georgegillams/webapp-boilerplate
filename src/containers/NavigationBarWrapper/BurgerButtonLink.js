import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';

const BurgerButtonLink = props => {
  const { href, ...rest } = props;

  return (
    <Link href={href}>
      <a {...rest} />
    </Link>
  );
};

BurgerButtonLink.propTypes = {
  href: PropTypes.string.isRequired,
};

export default BurgerButtonLink;
