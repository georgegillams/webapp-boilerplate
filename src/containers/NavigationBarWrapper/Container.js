import React from 'react';
import PropTypes from 'prop-types';
import { NavigationBar } from 'gg-components/NavigationBar';
import NavigationItem from 'components/NavigationItem';
import { SmallButtonSkeleton } from 'gg-components/Skeletons';

import Logo from 'components/Logo';

const NavigationBarWrapper = props => {
  const { authenticatorState } = props;
  const { user } = authenticatorState;

  const accountItem =
    user === undefined ? (
      <SmallButtonSkeleton />
    ) : (
      <NavigationItem name={user ? 'Account' : 'Login'} linkUrl={user ? '/account' : '/login'} />
    );

  const adminItem = user && user.admin ? <NavigationItem name="Admin" linkUrl="/admin" /> : null;

  const menuItems = [
    <NavigationItem key="home" name="Home" linkUrl="/" />,
    adminItem,
    accountItem,
    <NavigationItem key="sitemap" name="Site map" linkUrl="/sitemap" />,
  ];

  const date = new Date();
  const isPride = date.getMonth() === 5;

  return (
    <NavigationBar
      menuItems={menuItems}
      logo={<Logo padding={false} pride={isPride} animated />}
      accountMenuItem={accountItem}
    />
  );
};

NavigationBarWrapper.propTypes = {
  authenticatorState: PropTypes.shape({
    user: PropTypes.object,
  }).isRequired,
};

export default NavigationBarWrapper;
