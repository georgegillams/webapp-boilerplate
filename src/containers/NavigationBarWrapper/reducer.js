import { fromJS } from 'immutable';

const initialState = fromJS({});

function reducer(state = initialState) {
  return state;
}

export default reducer;
