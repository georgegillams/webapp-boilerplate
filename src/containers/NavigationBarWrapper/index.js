import { memo } from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import Container from './Container';

// import appSelectors from 'containers/App/selectors';
import { mapSelectors } from 'utils/redux-definitions/selectors';

const mapDispatchToProps = () => ({});

// const mapStateToProps = createStructuredSelector(mapSelectors(appSelectors));
const mapStateToProps = createStructuredSelector(mapSelectors([]));

const withConnect = connect(mapStateToProps, mapDispatchToProps);

export default compose(withConnect, memo)(Container);
