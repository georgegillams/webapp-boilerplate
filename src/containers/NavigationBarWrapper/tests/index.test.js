import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import configureStore from 'utils/redux/configure-store';

import NavigationBarWrapper from '../index';

describe('<Authenticator />', () => {
  let store;
  const state = {
    cookiesAllowed: false,
    loadingAuth: false,
    loadAuthError: null,
    user: undefined,
  };

  beforeAll(() => {
    store = configureStore({});
  });

  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <NavigationBarWrapper authenticatorState={state} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
