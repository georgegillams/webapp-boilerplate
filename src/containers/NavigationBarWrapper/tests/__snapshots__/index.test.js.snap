// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authenticator /> should render correctly 1`] = `
<div>
  <header
    role="banner"
  >
    <div
      class="navigation-bar__bar-placeholder"
    />
    <div
      class="navigation-bar__container"
    >
      <div
        aria-hidden="true"
        class="blur-effect__outer"
      />
      <div
        class="navigation-bar__bar"
      >
        <div
          class="navigation-bar__mobile-container navigation-bar__mobile-container--left"
        >
          <button
            aria-label="Menu"
            class="burger-button__outer navigation-bar__burger-button"
            type="button"
          >
            <div
              class="burger-button__inner"
            >
              <div
                class="burger-button__line burger-button__line__one navigation-bar__burger-button__line"
              />
              <div
                class="burger-button__line burger-button__line__two navigation-bar__burger-button__line"
              />
              <div
                class="burger-button__line burger-button__line__three navigation-bar__burger-button__line"
              />
            </div>
          </button>
        </div>
        <div
          class="navigation-bar__logo-container"
        >
          <div
            class="logo__container logo__container--no-padding"
          >
            <a
              href="/"
            >
              <h1
                aria-label="Home page"
                class="logo__heading logo__subheading--animated logo__heading--pride logo__heading--no-padding"
                role="button"
              >
                LOGO
              </h1>
            </a>
          </div>
        </div>
        <div
          class="navigation-bar__mobile-container navigation-bar__mobile-container--rgt"
        >
          <div
            class="skeleton__outer skeleton__button--small"
          >
            <div
              class="skeleton__shimmer"
              style="margin-left: -0px;"
            />
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
